packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -238,11 +256,14 @@ export const GridRootStyles = styled('div', {
       justifyContent: 'center',
       color: borderColor,
     },
-    [`& .${gridClasses.columnHeaders}:hover .${gridClasses.columnSeparator}, .${gridClasses['columnSeparator--resizing']}`]:
-      {
-        visibility: 'visible',
-        width: 'auto',
-      },
+    '@media (hover: hover)': {
+      [`& .${gridClasses.columnHeaders}:hover`]: columnHeadersStyles,
+      [`& .${gridClasses.columnHeader}:hover`]: columnHeaderStyles,
+    },
+    '@media (pointer: coarse)': {
```

Comment:
Is this one meant to be the opposite to the one above? It would be `@media (hover: none)` if I'm not mistaken.

Reply:
You're right